sdk/servicebus/azure-servicebus/api_review.py

```diff
@@ -0,0 +1,247 @@
+        logging_enable: bool = False,
+        http_proxy: dict = None,
+        transport_type: TransportType = None,
+        retry_total : int = 3,
+        retry_backoff_factor : float = 30,
+        retry_backoff_maximum : int = 120  # cur_retry_backoff = retry_backoff_factor * (2^cur_retry_time)
+    ) -> None:
+    @classmethod
+    def from_connection_string(
+        cls,
+        conn_str : str,
+        queue_name: str = None,
+        topic_name: str = None,
+        **kwargs
+    ) -> ServiceBusSender:
+
+    def __enter__(self):
+    def __exit__(self):
+
+    def close(self) -> None:
+
+    def get_properties(self) -> Dict[str, Any]:
+
+    def create_batch(
+        self,
+        max_size_in_bytes : int = None
+    ) -> BatchMessage:
+
+    def send(
+        self,
+        message : Union[Message, BatchMessage],
+        session_id : str = None,
+        message_timeout : float = None
+    ) -> None:
+    def schedule(
+        self,
+        message : Union[Message, BatchMessage],
+        schedule_time_utc : datetime,
+        session_id : str = None
+    ) -> List[int]:
+
+    def cancel_scheduled_messages(self, sequence_number : Union[int, List[int]]) -> None:
+
+
+class ServiceBusReceiver:
+    def __init__(
+        self,
+        fully_qualified_namespace : str,
+        credential: TokenCredential,
+        queue_name: str = None,
+        topic_name: str = None,
+        subscription_name : str = None,
+        session_id : str = None,
+        mode : ReceiveSettleMode = ReceiveSettleMode.PeekLock
+        logging_enable: bool = False,
+        http_proxy: dict = None,
+        transport_type: TransportType = None,
+        retry_total : int = 3,
+        retry_backoff_factor: float = 30,
+        retry_backoff_maximum: int = 120  # cur_retry_backoff = retry_backoff_factor * (2^cur_retry_time)
+    ) -> None:
+    @classmethod
+    def from_connection_string(
+        cls,
+        conn_str : str,
+        queue_name: str = None,
+        topic_name: str = None,
+        subscription_name : str = None,
+        session_id: str = None,
+        mode : ReceiveSettleMode = ReceiveSettleMode.PeekLock
+        **kwargs
+    )-> ServiceBusReceiver:
+
+    def __enter__(self):
+    def __exit__(self):
+
+    def close(self) -> None:
+
+    def get_properties(self) -> Dict[str, any]:
+
+    def __iter__(self):
+    def __next__(self):
+    def next(self):
+
+    def peek(
+        self,
+        count : int = 1,
+        start_from : int = None
+    ) -> List[PeekMessage]:
+    def receive_deferred_messages(
+        self,
+        sequence_numbers : List[int],
+    ) -> List[DeferredMessage]:
+    def receive(
+        self,
+        max_batch_size : int = None,
+        timeout : float = None,
+    ) -> List[ReceivedMessage]:  # Pull mode receive
+
+    @property
+    def session(self) -> ServiceBusSession:
+
+    # Rule APIs, raise Error when called on Queue
+    def add_rule(self, rule_name : str, filter : str) -> None:
```

Comment:
I believe this is out of scope for P1, but I'd be remiss to not mention my inclination to mirror the session encapsulation, error on accessing the property gives a better semantic signal to me, as well as containing this distinct functionality more cleanly from both a conceptual and visible surface area perspective.  (I already feel yak-shavey for even mentioning this, so feel free to just ignore.)